### PR TITLE
Add deployment, service and NOTES to create

### PR DESCRIPTION
**Pull request by viglesiasce**
This brings the `helm create` command templates up to date with what we expect to see in the Charts repo.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1461)

### CODE REVIEW

**Comment by viglesiasce** [comment]
Not exactly sure what are the right changes to make to get CI happy. Please advise.

**Comment by technosophos** [comment]
The simplify linter wants you to change a line:
```
pkg/chartutil/create.go:252:2:warning: 'if err != nil { return cdir, err }; return cdir, nil' can be simplified to 'return cdir, err' (gosimple)
```

I'll add a comment on the code.

File: pkg/chartutil/create.go [hunk 1]
```diff
@@ -120,5 +228,29 @@ func Create(chartfile *chart.Metadata, dir string) (string, error) {
...
+
+	// Write out NOTES.txt
+	val = []byte(defaultNotes)
+	if err := ioutil.WriteFile(filepath.Join(cdir, TemplatesDir, NotesName), val, 0644); err != nil {
+		return cdir, err
+	}
+
+	// Write out _helpers.tpl
+	val = []byte(defaultHelpers)
+	if err := ioutil.WriteFile(filepath.Join(cdir, TemplatesDir, HelpersName), val, 0644); err != nil {
```
Comment (technosophos) [inline pkg/chartutil/create.go]: The simplify linter points out that these four lines can be simplified to:

```
return ioutil.WriteFile(filepath.Join(cdir, TemplatesDir, HelpersName), val, 0644)
```

File: pkg/chartutil/create.go [hunk 2]
```diff
@@ -36,12 +36,37 @@ const (
 	ChartsDir = "charts"
 	// IgnorefileName is the name of the Helm ignore file.
 	IgnorefileName = ".helmignore"
+	// DeploymentName is the name of the example deployment file.
+	DeploymentName = "deployment.yaml"
+	// ServiceName is the name of the example service file.
+	ServiceName = "service.yaml"
```
Comment (viglesiasce) [inline pkg/chartutil/create.go]: I'd prefer to use the "long form", I'd rather forward port these behaviors than backport ones from existing charts. svc vs service seems like an unnecessary pre-optimization.

File: pkg/chartutil/create.go [hunk 3]
```diff
@@ -67,6 +92,89 @@ const defaultIgnore = `# Patterns to ignore when building packages.
...
+`
+
+const defaultNotes = `1. Get the application URL by running these commands:
+{{- if contains "NodePort" .Values.service.type }}
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT/login
+{{- else if contains "LoadBalancer" .Values.service.type }}
+**** NOTE: It may take a few minutes for the LoadBalancer IP to be available.                      ****
+****       You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}' ****
```
Comment (prydonius) [inline pkg/chartutil/create.go]: I would remove the '****' like we did in https://github.com/kubernetes/charts/blob/master/stable/drupal/templates/NOTES.txt#L15

File: pkg/chartutil/create.go (hunk 2 above)
Comment (prydonius) [inline pkg/chartutil/create.go]: Yeah, just wanted to point it out. I'm fine with either as long as there is a standard.

File: pkg/chartutil/create.go [hunk 4]
```diff
@@ -67,6 +92,89 @@ const defaultIgnore = `# Patterns to ignore when building packages.
...
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.Master.ServicePort }}
+{{- else if contains "ClusterIP"  .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "component={{ template "fullname" . }}-master" -o jsonpath="{.items[0].metadata.name}")
+  echo http://127.0.0.1:{{ .Values.service.externalPort }}
+  kubectl port-forward $POD_NAME {{ .Values.service.externalPort }}:{{ .Values.service.externalPort }}
+{{- end }}
+`
+
+const defaultHelpers = `{{/* vim: set filetype=mustache: */}}
```
Comment (rehevkor5) [inline pkg/chartutil/create.go]: This file is not really using Mustache, it's a Go Template, right?

**Pull request merged**